climada/util/coordinates.py

```diff
@@ -1057,6 +1057,79 @@ def assign_coordinates(coords, coords_to_assign, distance="euclidean",
                 coords_to_assign, coords[not_assigned_idx_mask], threshold, **kwargs)
     return assigned_idx
 
+def assign_haz_centroids(coord_df, hazard, crs, distance='euclidean',
+                        threshold=NEAREST_NEIGHBOR_THRESHOLD,
+                        overwrite=True):
+    """Assign to each exposure's coordinate point its closest hazard's coordinate.
+    If disatances > threshold in points' distances, -1 is returned.
+    If raster hazard and coordinate point outside raster, -1 is returned.
+
+    Parameters
+    ----------
+    coord_df : pd.DataFrame
+        Dataframe with defined latitude/longitude column
+    hazard : Hazard
+        Hazard to match (with raster or vector centroids).
+    crs : pyproj.crs.crs.CRS
+        Coordinate system of the dataframe
```

Comment:
why? if it's not the same as the one from `centroids` it just leads to an error?